R/retry.R

```diff
@@ -1,10 +1,27 @@
-retry <- function(expr, times = 3) {
-  tries <- 0
-  repeat {
-    tries <- tries + 1
-    if (tries >= times) { break }
-    outcome <- try(eval(expr), silent = TRUE)
-    if (methods::is(outcome, "try-error")) { next } else { break }
+#' Try evaluating an expression multiple times before erroring.
+#'
+#' @param num_tries numeric. The number of attempts.
+#' @param sleep_secs numeric. How long to wait between attempts.
+#' @param expr expression. The expression to evaluate with retries.
+#' @return TRUE if the expression gets evaluated successfully on some attempt.
+#' @export
+with_retries <- function(expr, num_tries = 1, sleep_secs = 0.001) {
+  num_tries <- max(num_tries, 1L)
+  current_try <- 0
+  while (current_try < num_tries) {
+    out <- evaluate::try_capture_stack(expr, env = parent.frame())
```

Comment:
Is `evaluate` a base package or does it need to be in `DESCRIPTION`? What's the advantage of `try_capture_stack` over `try`?